# SPDX-FileCopyrightText: 2025 coolGi <me@coolgi.dev>
#
# SPDX-License-Identifier: GPL-3.0-only

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 62c6188..e586ac8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,57 +25,11 @@ option(KAON_OPENXR "OpenXR headset view for WiVRn and Monado" ON)
 # SteamVR is not installed. The stub is fetched as individual files to avoid
 # cloning the whole SDK.
 if(KAON_OPENVR)
-set(OPENVR_VERSION v2.15.6)
-set(OPENVR_SRC_DIR ${CMAKE_BINARY_DIR}/_deps/openvr-stub-src)
-set(OPENVR_FILES
-    headers/openvr.h
-    src/openvr_api_public.cpp
-    src/ivrclientcore.h
-    src/jsoncpp.cpp
-    src/json/json.h
-    src/json/json-forwards.h
-    src/vrcore/dirtools_public.cpp
-    src/vrcore/dirtools_public.h
-    src/vrcore/envvartools_public.cpp
-    src/vrcore/envvartools_public.h
-    src/vrcore/hmderrors_public.cpp
-    src/vrcore/hmderrors_public.h
-    src/vrcore/pathtools_public.cpp
-    src/vrcore/pathtools_public.h
-    src/vrcore/sharedlibtools_public.cpp
-    src/vrcore/sharedlibtools_public.h
-    src/vrcore/strtools_public.cpp
-    src/vrcore/strtools_public.h
-    src/vrcore/vrpathregistry_public.cpp
-    src/vrcore/vrpathregistry_public.h)
-foreach(OPENVR_FILE IN LISTS OPENVR_FILES)
-    if(NOT EXISTS ${OPENVR_SRC_DIR}/${OPENVR_FILE})
-        message(STATUS "Downloading openvr ${OPENVR_FILE}")
-        file(DOWNLOAD
-            https://raw.githubusercontent.com/ValveSoftware/openvr/${OPENVR_VERSION}/${OPENVR_FILE}
-            ${OPENVR_SRC_DIR}/${OPENVR_FILE}
-            SHOW_PROGRESS)
-    endif()
-endforeach()
-add_library(openvr_api STATIC
-    ${OPENVR_SRC_DIR}/src/openvr_api_public.cpp
-    ${OPENVR_SRC_DIR}/src/jsoncpp.cpp
-    ${OPENVR_SRC_DIR}/src/vrcore/dirtools_public.cpp
-    ${OPENVR_SRC_DIR}/src/vrcore/envvartools_public.cpp
-    ${OPENVR_SRC_DIR}/src/vrcore/hmderrors_public.cpp
-    ${OPENVR_SRC_DIR}/src/vrcore/pathtools_public.cpp
-    ${OPENVR_SRC_DIR}/src/vrcore/sharedlibtools_public.cpp
-    ${OPENVR_SRC_DIR}/src/vrcore/strtools_public.cpp
-    ${OPENVR_SRC_DIR}/src/vrcore/vrpathregistry_public.cpp)
-target_include_directories(openvr_api PUBLIC
-    ${OPENVR_SRC_DIR}/headers
-    ${OPENVR_SRC_DIR}/src
-    ${OPENVR_SRC_DIR}/src/vrcore)
-if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|ARM64")
-    target_compile_definitions(openvr_api PRIVATE LINUX POSIX LINUXARM64 VR_API_PUBLIC VRCORE_NO_PLATFORM)
-else()
-    target_compile_definitions(openvr_api PRIVATE LINUX POSIX LINUX64 VR_API_PUBLIC VRCORE_NO_PLATFORM)
-endif()
+find_library(OPENVR_LIBRARIES NAMES openvr_api)
+find_path(OPENVR_INCLUDE_DIR openvr/openvr.h)
+add_library(openvr_api SHARED IMPORTED)
+set_property(TARGET openvr_api PROPERTY IMPORTED_LOCATION ${OPENVR_LIBRARIES})
+target_include_directories(openvr_api INTERFACE ${OPENVR_INCLUDE_DIR})
 endif()
 
 add_subdirectory(src)
diff --git a/src/VrOverlay.h b/src/VrOverlay.h
index 14edb91..26d51ef 100644
--- a/src/VrOverlay.h
+++ b/src/VrOverlay.h
@@ -11,7 +11,7 @@
 #include <memory>
 
 #if defined(KAON_OPENVR)
-    #include <openvr.h>
+    #include <openvr/openvr.h>
 #endif
 
 #if defined(KAON_OPENXR)
